scripts/metricsgen/metricsdiff/metricsdiff_test.go

```diff
@@ -0,0 +1,64 @@
+package main_test
+
+import (
+	"bytes"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	metricsdiff "github.com/tendermint/tendermint/scripts/metricsgen/metricsdiff"
+)
+
+func TestDiff(t *testing.T) {
+	for _, tc := range []struct {
+		name      string
+		aContents string
+		bContents string
+
+		expectedDiff metricsdiff.MetricsDiff
+	}{
+		{
+			name: "labels",
+			aContents: `
+			metric_one{label_one="content", label_two="content"} 0
+			`,
+			bContents: `
+			metric_one{label_three="content", label_four="content"} 0
+			`,
+			expectedDiff: metricsdiff.MetricsDiff{
```

Comment:
This is totally reasonable, but I am lazy and would probably have just compared the rendered diff output since that requires the struct to have the right shape and tests both (I mention this only because there were some missing newlines in the string render that weren't caught here—YMMV).

Reply:
Done.